### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by jerry1ye10**
Added comments feature into the api by using a scaffold.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    elsif params[:user_id]
+      @comments = User.friendly.find(params[:user_id]).comments
+    end
+    render json: @comments
+  end
+
+  # GET /comments/1
+  def show
+    if params[:article_id]
+      if @comment.article_id == params[:article_id]
```
Comment (NicholasLYang) [inline app/controllers/comments_controller.rb]: consider using postfix if. Very useful

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  private
+  # Use callbacks to share common setup or constraints between actions.
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  # Only allow a trusted parameter "white list" through.
+  def comment_params
+    params.require(:comment).permit(:content, :text, :comment_id, :user_id, :article_id)
```
Comment (NicholasLYang) [inline app/controllers/comments_controller.rb]: 80 col rule

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  # Use callbacks to share common setup or constraints between actions.
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  # Only allow a trusted parameter "white list" through.
+  def comment_params
+    params.require(:comment).permit(:content,
+                                    :text, :comment_id, :user_id, :article_id)
+  end
```
Comment (jkao1) [inline app/controllers/comments_controller.rb]: Why does the comment class need params like its own id? Other classes don't have it.

File: config/routes.rb [hunk 4]
```diff
@@ -1,8 +1,11 @@
 Rails.application.routes.draw do
+  resources :comments
   resources :media
   resources :authorships
   mount_devise_token_auth_for 'User', at: 'auth'
-  resources :users
+  resources :users do
+    resources :comments
```
Comment (jkao1) [inline config/routes.rb]: Remove this nested route. We'll add it back when we actually need it. But in the actual CFA, we load everything literally (except articles) so I don't see myself using this.

File: config/routes.rb [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
+  end
   resources :sections do
     resources :articles do
       resources :media
```
Comment (jkao1) [inline config/routes.rb]: Triple nested route haysus crysto. Remove the media route underneath articles.

File: config/routes.rb (hunk 4 above)
Comment (jkao1) [reply config/routes.rb]: Moreover, for CUD operations on comments, I'm already doing that through articles, never through users.

File: config/routes.rb (hunk 5 above)
Comment (jkao1) [reply config/routes.rb]: Remind me to talk about this in tomorrow's meeting. We need to actually think about this and remove all unnecessary routes.

**Pull request merged**